compiler/src/iree/compiler/Codegen/Common/TransformDialectInterpreterPass.cpp

```diff
@@ -104,14 +108,40 @@ static Operation *findOpWithTag(Operation *root, StringRef tagKey,
     // In debug mode, continue the traversal to see if the tag is not
     // duplicated.
 #ifndef NDEBUG
-    return WalkResult::advance();
-#else
     return WalkResult::interrupt();
+#else
+    return WalkResult::advance();
```

Comment:
I don't know what this is doing, but whenever I see something like this in production code, I get itchy: functional changes based on debug mode are pretty verboten.

I *think* what you are trying to do is build a debug facility of some kind that lets you inspect/alter the flow, which seems reasonable. But that should be separated in some way from the regular execution flow, and it should be guarded in a more robust way than an inline functionality switch like this in the main code path.